migrations/env.py

```diff
@@ -63,13 +63,25 @@ def run_migrations_online():
     and associate a connection with the context.
 
     """
+    # abort any migrations if the lock cannot be acquired after one second.
+    #
+    # if we see issues with this lock timeout failing, we should try running again when there are no locks on that
+    # table, perhaps at a quieter time.
+    options = {"lock_timeout": "1000"}
```

Comment:
Do I understand well that it won't be able to acquire a lock if a SELECT is running?

If so, any big service user looking at their dashboard would cause that to fail, right? 

I like the idea in general, it's a great safety measure. At the same time 1 second sounds pretty short - I see how the long we wait, the bigger potential query buildup. Happy to go with this as if we run into trouble we can easily reverse it / extend the lock timeout time.

[pondering] I wonder if this could affect our upcoming whole database migration? Probably not, since we will run the migrations before we start populating the db?

Reply:
Yeah that's right. I think if we run into issues with this we'll want to look at wrapping the migration in a retry loop (so if we get a `LockNotAvailable` we retry, eg, five times before failling)

Reply:
I don't think this'll affect our whole DB migration as that's using AWS DMS rather than our alembic scripts I believe?